easybuild/tools/module_generator.py

```diff
@@ -783,10 +783,12 @@ def load_module(self, mod_name, recursive_unload=False, unload_modules=None):
         body.append(self.LOAD_TEMPLATE)
 
         if build_option('recursive_mod_unload') or recursive_unload:
-            # not wrapping the 'module load' with an is-loaded guard ensures recursive unloading;
-            # when "module unload" is called on the module in which the depedency "module load" is present,
-            # it will get translated to "module unload"
-            load_statement = body + ['']
+            # wrapping the 'module load' with an 'is-loaded or mode == unload'
+            # guard ensures recursive unloading while avoiding load storms,
+            # when "module unload" is called on the module in which the
+            # depedency "module load" is present, it will get translated
+            # to "module unload"
+            load_statement = [self.conditional_statement('isloaded("%(mod_name)s") or mode() == "unload"', '\n'.join(body), negative=True)]
         else:
             load_statement = [self.conditional_statement('isloaded("%(mod_name)s")', '\n'.join(body), negative=True)]
```

Comment:
@akesandgren This is a bit too copy-pasta, let's change this too the below, and include a reference to http://lmod.readthedocs.io/en/latest/210_load_storms.html:

```python
if build_option('recursive_mod_unload') or recursive_unload:
    # ...
    # see also http://lmod.readthedocs.io/en/latest/210_load_storms.html
    load_guard = isloaded("%(mod_name)s") or mode() == "unload"'
else:
    load_guard = 'isloaded("%(mod_name)s")'
load_statement = [self.conditional_statement(load_guard, '\n'.join(body), negative=True)]